### PR TITLE
ja: Translate Chapter 54, Control Flow

### PR DESCRIPTION
**Pull request by kantasv**
Hi, JA project folks (#652), here's a MR for the chapter "Control Flow". When you have some time, could you review the diff? Any feedback or suggestions are greatly appreciated. Thank you in advance.

### CODE REVIEW

File: po/ja.po [hunk 1]
```diff
@@ -17976,12 +17981,18 @@ msgid ""
 "you must know how many futures you will have at compile time. This is "
 "currently in the `futures` crate, soon to be stabilised in `std::future`."
 msgstr ""
+"複数の互いに素な型のfutureに対しては、`std::future::join!`を利用できます。し"
+"かし、いくつのfutureがコンパイル時に存在しているのかを把握しておく必要があり"
+"ます。これは現在`future`クレートにありますが、近いうちに`std::future`に統合さ"
```
Comment (HidenoriKobayashi) [inline po/ja.po:17986]: nit: the name of the crate seems to be with 's', so `futures` 😄

File: po/ja.po [hunk 2]
```diff
@@ -17976,12 +17981,18 @@ msgid ""
...
+"れる予定です。"
 
 #: src/async/control-flow/join.md:42
 msgid ""
 "The risk of `join` is that one of the futures may never resolve, this would "
 "cause your program to stall. "
 msgstr ""
+"`join`の持つリスクは、複数のfutureのうちの１つが解決されないことです。この場"
+"合、プログラムは停止してしまいます。 "
```
Comment (HidenoriKobayashi) [inline po/ja.po:17995]: Could we simply say something like below?
```suggestion
"`join`のリスクは、複数のfutureのうちの１つでも解決されないと"
"プログラムがストールしてしまうということです。 "
```

i tried to be more explicit about the "may" in the original sentence by phrasing it like "if". Also I left "stall" in Katakana as that is a technical term, I guess.

File: po/ja.po [hunk 3]
```diff
@@ -18065,30 +18089,42 @@ msgid ""
...
+"うした枝のfutureがキャンセルされることにつながります。`select!`を毎回実行する"
+"際に新たなfutureが作成されるときに、`select!`を使うのが最も簡単です。"
 
 #: src/async/control-flow/select.md:76
 msgid ""
 "An alternative is to pass `&mut future` instead of the future itself, but "
 "this can lead to issues, further discussed in the pinning slide."
 msgstr ""
+"Futureそのものでなく、`&mut future`を渡すという代替案もあります。しかし、これ"
+"は問題につながることもあります。このことは固定スライドで詳細に議論します。"
```
Comment (HidenoriKobayashi) [inline po/ja.po:18127]: hmm, if I'm not mistaken "the pinning slide" refers to https://google.github.io/comprehensive-rust/async/pitfalls/pin.html, right?
I think the term "pin" should not be translated, in that case.

So for the last sentence, I would put it like below
```suggestion
"は問題につながることもあります。このことはPinに関するスライドで詳細に議論します。"
```

File: po/ja.po (hunk 2 above)
Comment (kantasv) [reply po/ja.po:17995]: Yeah, "ストール" captures more specific type of failures than "停止" delivers. I totally agree with you, and I'll add it.

File: po/ja.po (hunk 1 above)
Comment (kantasv) [reply po/ja.po:17986]: Thanks! Seems like I dropped `s` without noticing.. :)

**Comment by kantasv** [comment]
Thanks @HidenoriKobayashi for suggestions and feedback. I've added a commit to apply them. For JA translation folks, could you review the MR? Arigato-gozaimasu in advance.

File: po/ja.po [hunk 4]
```diff
@@ -17999,6 +18015,10 @@ msgid ""
 "race`. In Python, it compares to `asyncio.wait(task_set, return_when=asyncio."
 "FIRST_COMPLETED)`."
 msgstr ""
+"Selectという操作では、futureの集合うち、いずれか１つの準備が整うまで待機し、"
```
Comment (keiichiw) [inline po/ja.po:18018]:
```suggestion
"Selectという操作では、futureの集合のうち、いずれか１つの準備が整うまで待機し、"
```

File: po/ja.po [hunk 5]
```diff
@@ -18065,30 +18089,43 @@ msgid ""
 "whichever arrives first. Since the dog takes 50ms, it wins against the cat "
 "that take 500ms seconds."
 msgstr ""
+"この例では、猫と犬のレースを扱っています。`first_animal_to_finish_race`は両方"
+"のチャネルをリッスンし、先に到着した方を選びます。犬は到着まで50msかかるの"
+"で、100msかかる猫に勝ちます。"
```
Comment (keiichiw) [inline po/ja.po:18094]:
```suggestion
"で、500msかかる猫に勝ちます。"
```

File: po/ja.po [hunk 6]
```diff
@@ -18065,30 +18089,43 @@ msgid ""
...
 msgstr ""
+"レースに制限時間を追加することによって、違う種類のfutureをselectすることを実"
+"演してみてください。"
 
 #: src/async/control-flow/select.md:73
 msgid ""
 "Note that `select!` drops unmatched branches, which cancels their futures. "
 "It is easiest to use when every execution of `select!` creates new futures."
 msgstr ""
+"`select!`はマッチしなかった枝をドロップすることに注意してください。これは、そ"
```
Comment (keiichiw) [inline po/ja.po:18117]:
```suggestion
"`select!`はマッチしなかったブランチをドロップすることに注意してください。これは、そ"
```

File: po/ja.po [hunk 7]
```diff
@@ -18065,30 +18089,43 @@ msgid ""
...
+"うした枝のfutureがキャンセルされることにつながります。`select!`を毎回実行する"
```
Comment (keiichiw) [inline po/ja.po:18118]:
```suggestion
"うしたブランチのfutureがキャンセルされることにつながります。`select!`を毎回実行する"
```

File: po/ja.po (hunk 6 above)
Comment (keiichiw) [reply po/ja.po:18117]: I think it's not so common to translate this "branch" to "枝"

**Review by keiichiw — APPROVED** [review]
LGTM with minor suggestions

**Comment by kantasv** [comment]
Thank you @keiichiw for reviewing the diff and providing concrete suggestions. I have incorporated your suggestions. Please merge the MR if it looks okay to you :)

**Comment by keiichiw** [comment]
Thank you!

**Pull request merged**